src/shared/components/ShareThroughEmailModal/ShareThroughEmailModal.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FunctionComponent, useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+
+import {
+	BlockHeading,
+	Box,
+	Button,
+	Flex,
+	FlexItem,
+	Modal,
+	ModalBody,
+	Spacer,
+	Spinner,
+	TextInput,
+} from '@viaa/avo2-components';
+
+import toastService from '../../../shared/services/toast-service';
+import { copyToClipboard } from '../../helpers';
+import { shareThroughEmail } from '../../helpers/share-through-email';
+
+import './ShareThroughEmailModal.scss';
+
+interface AddToCollectionModalProps {
+	modalTitle: string;
+	type: 'item' | 'collection' | 'bundle';
+	emailLinkHref: string;
+	emailLinkTitle: string;
+	isOpen: boolean;
+	onClose: () => void;
+}
+
+const ShareThroughEmailModal: FunctionComponent<AddToCollectionModalProps> = ({
+	modalTitle,
+	type,
+	emailLinkHref,
+	emailLinkTitle,
+	isOpen,
+	onClose,
+}) => {
+	const [t] = useTranslation();
+
+	const [emailAddress, setEmailAddress] = useState<string>('');
+	const [isProcessing, setIsProcessing] = useState<boolean>(false);
+
+	const sendEmail = async () => {
+		try {
+			setIsProcessing(true);
+			await shareThroughEmail(emailAddress, emailLinkTitle, emailLinkHref, type);
+			toastService.success(t('De email is verstuurd'));
+		} catch (err) {
+			console.error('Failed to send email to share item', err, {
+				emailAddress,
+				emailLinkTitle,
+				emailLinkHref,
+				type,
+			});
+			toastService.danger('Het versturen van de email is mislukt');
+		}
+		setIsProcessing(false);
+	};
+
+	return (
+		<Modal className="m-share-through-email-modal" title={modalTitle} size="medium" isOpen={isOpen} onClose={onClose} scrollable>
+			<ModalBody>
+				<BlockHeading type="h4">
+					<Trans>Kopieer deze publieke link</Trans>
+				</BlockHeading>
+				<p>
+					<Trans>
+						Let wel, enkel personen met een Archief voor Onderwijs account zullen deze link kunnen
+						openen.
+					</Trans>
+				</p>
+				<Spacer margin="top-large">
+					<Box backgroundColor="gray" condensed>
+						<Flex wrap justify="between" align="baseline">
+							<FlexItem className="c-ellipsis">
+								<a href={emailLinkHref}>{emailLinkHref}</a>
+							</FlexItem>
+							<FlexItem shrink>
+								<Spacer margin="left-small">
+									<Button
+										label={t('Kopieer link')}
+										onClick={() => {
```

Comment:
Ik zou deze functie eruit trekken om wat beter overzicht te hebben in uw template. Zou ook voor consistency zorgen want paar lijnen hieronder doe je het wel. :)